basex-core/src/main/java/org/basex/query/func/sql/SqlExecute.java

```diff
@@ -39,7 +38,7 @@ public Iter iter(final QueryContext qc) throws QueryException {
     if(!(obj instanceof Connection)) throw BXSQ_CONN_X.get(info, id);
     try {
       final Statement stmt = ((Connection) obj).createStatement();
-      return stmt.execute(query) ? iter(stmt, true) : Empty.ITER;
+      return stmt.execute(query) ? iter(stmt, true) : Int.get(stmt.getUpdateCount()).iter();
```

Comment:
Ok, I did some further investigation and tests on oracle, mysql and postgresql.

If a SELECT statement was processed by execute, it will be always return true, even no result was returned ("select * from sometable where 1=0"). So an empty sequence will return within xquery anyway and not change in current behaviour of BaseX. 

If it was an update (insert, delete or ddl) statement, execute returns always false and the number of affected record can received by getUpdateCount. getUpdateCount will return 0 if no rows are affected ("update someTable set someColumn='x' where 1=0"). 

In consequence it will not mixed up with empty sequence as result of an empty select.

So the patch should be fine as it is :)